### ISSUE TITLE
Docs: add "Advanced install" / "Setting up Docker for production" section

### ISSUE THREAD

**Issue by thaJeztah** [moby/moby#15946]
We currently have https://get.docker.com to get users up and running in "no" time, so that they can play around with Docker. Let's call it "express install".

For Linux, this is https://get.docker.com, for Windows and Mac users, we have Docker Toolbox, which (to me) fall in the same category; get users started quickly.

However, the "express install" is in many cases _not_ suitable for production; we make many assumptions, and some settings we use are just to "get it working", but require manual configuration to be really useful (for example, the default device mapper configuration is not suitable for production).

We should point users to an "advanced install" section in the documentation, and _warn_ users that the "express install" should be reviewed before using in production.

The advanced install section in the documentation, should cover (or link to) - this is not a full list, just some initial thoughts;
- Checking prerequisites (e.g., using `/contrib/check-config.sh`)
- Pinning to a specific version of Docker
- Picking, setting up, and configuring a graph driver (e.g. Setting up aufs)
- Configuring daemon options (SysVinit, SystemD)
- Picking, and configuring a default logging-driver
- Securing Docker (Certificates, AppArmor, SeLinux)
- Networking options and related (e.g. Certificates, Userland Proxy, FireWall issues)

TL;DR; we should make it as easy as possible for people to get to "play" with Docker, but equally (if not more) important, we should explain that Docker uses advanced techniques; making optimum use of Docker _does_ require careful planning and configuration, and making informed decisions (e.g. The best graph driver for your situation).

**Comment by ahawkins** [comment]
@thaJeztah I was thinking this myself! There is some documentation currently scattered around the docker doc website. I wanted this sort of into trying to create a production UCP cluster. Ended up going to the docker-engine docs to see how to configure the most production ready storage driver.

**Comment by thaJeztah** [comment]
There's quite a lot of changed made to the documentation, including separate instructions for Docker EE, and Docker CE, so I' think we should close this for now
